### PR TITLE
wr-60: Edit User Profile functionality

### PR DESCRIPTION
**Pull request by drMangust**
Frontend (FE):

- Implemented a "Edit profile" button that toggles between non-editable and editable modes.
- Designed an editable form with input fields for first name, last name, email, and profile picture.
- Prepopulated the form with accurate existing user information.
- Enabled users to modify any or all fields in the edit form.
- Implemented email field validation to accept valid email formats.
- Displayed error messages and highlighted fields with invalid email or name inputs.
- Included "Cancel" and "Save" buttons for reverting changes or saving edits.
- Implemented data validation on clicking the "Save" button.
- After successful validation, switched back to non-editable mode with updated information.

Backend (BE):

- Created a route to handle PUT requests for updating user data (/users).
- Implemented shared validation logic for both frontend and backend.
- Added a repository/service method for updating user data.
- Introduced a UserUpdateRequestDto and userUpdateValidationSchema.
- Extended types for "passwordHash" and "passwordSalt" in UserEntity.
- Added useful fields to the object returned by useAppForm.
- Enhanced the logic for the getCurrentUser action.
- Updated user repository methods and changed access wrappers for some routers.
- Created an ExceptionMessage enum for handling exceptions.

### CODE REVIEW

File: frontend/src/pages/profile/profile.tsx [hunk 1]
```diff
@@ -12,13 +23,24 @@ const mockedUser = {
 };
 
 const Profile: React.FC = () => {
+  const [editMode, setEditMode] = useState(false);
```
Comment (andanf-e) [inline frontend/src/pages/profile/profile.tsx]:
```suggestion
  const [isEditingProfile, setIsEditingProfile] = useState(false);
```

File: frontend/src/slices/auth/actions.ts [hunk 2]
```diff
@@ -38,12 +38,26 @@ const signIn = createAsyncThunk<
...
+
+  const token = await storage.get(StorageKey.TOKEN);
+  const hasToken = Boolean(token);
+
+  if (!hasToken) {
+    return null;
+  }
+
+  try {
+    return await authApi.getCurentUser();
```
Comment (andanf-e) [inline frontend/src/slices/auth/actions.ts]: Missed typo before :c
```suggestion
    return await authApi.getCurrentUser();
```

Comment (drMangust) [reply frontend/src/slices/auth/actions.ts]: fix - [6bb6359](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/94/commits/6bb63592cd3477979337716a326a29c8e06b3f8a)

File: frontend/src/pages/profile/profile.tsx (hunk 1 above)
Comment (drMangust) [reply frontend/src/pages/profile/profile.tsx]: fix - [9350dd6](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/94/commits/9350dd658729d04311a8148cddf8410d5c8dcafc)

**Review by andanf-e — APPROVED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Pull request merged**